File: mns-core/src/main/java/cz/fb/manaus/core/category/categorizer/TypeCategorizer.java
package cz.fb.manaus.core.category.categorizer;

import cz.fb.manaus.core.model.Market;
import org.springframework.stereotype.Component;

import java.util.Optional;
import java.util.Set;

@Component
final public class TypeCategorizer extends AbstractDelegatingCategorizer {


    public static final String PREFIX = "type_";

    public TypeCategorizer() {
        super(PREFIX);
    }

    @Override
    public Set<String> getCategoryRaw(Market market) {
        var type = market.getType();
        return Set.of(Optional.ofNullable(type).map(String::toLowerCase).orElse("unknown"));

    }
}

File: mns-core/src/main/java/cz/fb/manaus/core/category/categorizer/TypeCategorizer.kt
package cz.fb.manaus.core.category.categorizer

import cz.fb.manaus.core.model.Market
import org.springframework.stereotype.Component

@Component
object TypeCategorizer : AbstractDelegatingCategorizer("type_") {

    public override fun getCategoryRaw(market: Market): Set<String> {
        return when (market.type) {
            null -> emptySet()
            else -> setOf(market.type.lowercase())
        }
    }
}
